path_roles.go

```diff
@@ -345,13 +345,18 @@ func (b *azureSecretBackend) createPersistedApp(ctx context.Context, req *logica
 		return err
 	}
 
+	assignmentIDs, err := c.generateUUIDs(len(role.AzureRoles))
+	if err != nil {
+		return fmt.Errorf("error generating assginment IDs; err=%w", err)
+	}
+
 	if role.ManagedApplicationObjectID != "" {
 		removeRolesAndGroupMembership(ctx, c, role)
 
 		spObjID := role.ServicePrincipalObjectID
 
 		// Assign Azure roles to the new SP
-		raIDs, err := c.assignRoles(ctx, spObjID, role.AzureRoles)
+		raIDs, err := c.assignRoles(ctx, spObjID, role.AzureRoles, assignmentIDs)
```

Comment:
Related to how this integrates with the persisted app feature, I think this means that partial failure here wouldn't clean up role assignments? We have a mechanism for doing so introduced by https://github.com/hashicorp/vault-plugin-secrets-azure/pull/110. We should create a task to account for this in persisted app creation. We should be able to write this similar to what's done in https://github.com/hashicorp/vault-plugin-secrets-azure/blob/main/path_service_principal.go#L124-L144.